### ISSUE TITLE
Is there a cast function?

### ISSUE THREAD

**Issue by xermao** [20tab/UnrealEnginePython#701]
I have a function in bluerpint return a SkeletalMeshComponent,but in python,it recognition as a uobject.
So how can I cast it to a SkeletalMeshComponent?

**Comment by dfb** [comment]
I'm not a dev of this project, but there isn't a cast function that I know of, but there also isn't a need for one. What you see in Python - the uobject - is a Python wrapper around an engine UObject. You can use the 'is_a' function to test if the thing being wrapped is a particular type of object:

```py
from unreal_engine.classes import SkeletalMeshComponent
your_uobject.is_a(SkeletalMeshComponent)
```

(I can't test the above out at the moment, so it might not be 100% correct code, but that's the basic idea)

Some class-specific APIs are exposed on the Python wrapper object, but beyond that there are many dynamic APIs (basically: if the C++ header file defines something as 'BlueprintCallable', then you can call it from Python). See [the uobject API](https://github.com/20tab/UnrealEnginePython/blob/master/docs/uobject_API.md) for more info.

**Comment by kogen00** [comment]
Unfortunately the method provided here does not work for UObject to Blueprints.
What I currently know is that this does not work:

`
import unreal_engine as ue
from unreal_engine.classes import ActorFactory, BlueprintFactory, DirectionalLightComponent, K2Node_Event
bpFactory = BlueprintFactory()
bp = ue.find_object('/Game/CarAgent')
bp.is_a(BlueprintFactory)
`
is_a is not a cast it simpy checks if your Object is a Blueprintfactory in this case. So the question is still valid how do you cast a UObject to something else?

**Comment by dfb** [comment]
Yes, that's what I said - is_a lets you know if a cast would have made sense. In this case, /Game/CarAgent is not a BlueprintFactory, so it would not have made sense to cast it anyway.

To reiterate, I'm not aware of any case where you *need* to cast - that concept doesn't really translate to Python (Python is a strongly but dynamically typed language, casting is needed in strongly but statically typed languages like C++).

So in this specific example, what type of asset/object is /Game/CarAgent? (what does it subclass?) It apparently doesn't subclass UBlueprintFactory, which is both why is_a returns False but also why it wouldn't make sense to try and cast it to one.

If you can provide more details on what you're trying to accomplish, I might be able to help you. I'm pretty sure casting is not what you need though.

**Comment by kogen00** [comment]
Sorry, after I posted I read it again and realized that's what you said, then I wanted to change it and got distracted with a different task.
Anyways, I found a solution where the operator checked for an asset at the asset path, and if found destroyed the asset. That's one way of doing it and I'm pretty sure that's how I will proceed -> all or nothing.
But if you created a Blueprint and you then want to add a component to this blueprint it does not work without a cast.
I mean: Create a blueprint if the object is not existing, add a component.
If the object exists, get the blueprint from the found UObject (here I would need the cast) and add the component.
If you just use the UObject Unreal python from 20tabs will simply throw an error that the object where you try to add the component to is a UObject and needs to be a UBlueprintFactory.
